Mathlib/CategoryTheory/NatIso.lean

```diff
@@ -0,0 +1,266 @@
+/-
+Copyright (c) 2017 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tim Baumann, Stephen Morgan, Scott Morrison, Floris van Doorn
+-/
+import Mathlib.CategoryTheory.Functor.Category
+import Mathlib.CategoryTheory.Iso
+
+/-!
+# Natural isomorphisms
+
+For the most part, natural isomorphisms are just another sort of isomorphism.
+
+We provide some special support for extracting components:
+* if `α : F ≅ G`, then `a.app X : F.obj X ≅ G.obj X`,
+and building natural isomorphisms from components:
+*
+```
+nat_iso.of_components
+  (app : ∀ X : C, F.obj X ≅ G.obj X)
+  (naturality : ∀ {X Y : C} (f : X ⟶ Y), F.map f ≫ (app Y).hom = (app X).hom ≫ G.map f) :
+F ≅ G
+```
+only needing to check naturality in one direction.
+
+## Implementation
+
+Note that `nat_iso` is a namespace without a corresponding definition;
+we put some declarations that are specifically about natural isomorphisms in the `iso`
+namespace so that they are available using dot notation.
```

Comment:
```suggestion
Note that `NatIso` is a namespace without a corresponding definition;
we put some declarations that are specifically about natural isomorphisms in the `Iso`
namespace so that they are available using dot notation.
```